auto_gen_tool.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import os
+import re
+import sys
+
+
+def add_new_line(file, line_num, text):
+    with open(file, "r") as file_handler:
+        lines = file_handler.readlines()
+        if (line_num < 0) or (line_num > len(lines) + 1):
+            return False
+
+    lines.insert(line_num - 1, text + os.linesep)
+    with open(file, "w") as file_handler:
+        for line in lines:
+            file_handler.write(line)
+
+    return True
+
+
+def add_package_declarations(generated_root_path):
+    file_names = os.listdir(generated_root_path)
+    for file_name in file_names:
+        if not file_name.endswith('.java'):
+            break
+        full_name = generated_root_path + '/' + file_name
+        add_new_line(full_name, 2, 'package org.ray.runtime.generated;')
+
+
+def get_offset(file, field):
+    with open(file, "r") as file_handler:
+        lines = file_handler.readlines()
+
+        for line in lines:
+            re_str = '.*%s\(int j\) \{ int o = __offset\((.*)\);.*' % field
+            results = re.findall(re_str, line)
+            if len(results) == 0:
+                continue
+            elif len(results) == 1:
+                return int(results[0])
+            else:
+                return -1
+
+        return -1
+
+
+template = '''
```

Comment:
`template` looks too general here. Shall we use `template_for_byte_buffer_getter`?